### ISSUE TITLE
SVG maps too heavy as include external vector points

### ISSUE THREAD

**Issue by CarloCocozza** [Pandora-IsoMemo/DSSM#259]
Whenever svg maps from TimeR or AverageR are exported, these also include vector points covering the whole globe. This renders each file extremely heavy when edited (e.g. in Inkscape). Unfortunately I cannot attach an example map as the file is too heavy. Please see its PNG version and a screenshot on inkscape.

![Screenshot 2024-11-29 213930](https://github.com/user-attachments/assets/027281a2-c272-4604-b3c4-ef53078d3662)
![spatio-temporal-average_900](https://github.com/user-attachments/assets/8e1d467e-da12-4170-8bd7-b4e8498ec84e)

I also attach the used dataset and I copy-paste the parameters I used in TimeR. I have exported svg files from the gif option.

[Filtered Database.xlsx](https://github.com/user-attachments/files/17962860/Filtered.Database.xlsx)

Dependent variable: IRMS.D13C.coll
Dependent variable type: numeric
Uncertainty (optional of dep. var.: IRMS.D13C.coll.unc
Date type: Interval
Mean or lower time point: Min. Year 95
Standard deviation or upper time point: Max. Year 95
Longitude variable: Longitude
Latitude variable: Latitude
Site/ID (optional): Site.name
Smooth type: planar 
Center of data: 0th Meridian
Border Correction for Pacific: NO
Number of basic functions: 250 
Extrapolation behavior: Constant
Remove model outliers: NO
Remove data outliers: NO
Restrict model area: NO
Bayesian model: YES
Varying standard error: NO 
Distribution of date uncertainty: Uniform (full width)
Number of MCMC iterations: 10000 
Number of burnin iterations: 3000
Number of MCMC chains: 1 
MCMC thinning (keep every x-th sample): 10 
Amount of smoothing: 1 

AFTER MODELLING

Map Centering: 0th meridian 
Estimation type: Mean 
Show model estimates: YES 
Range of colour scale 
•	Max range: -14
•	Min range:  -21
Range restriction: No restriction 
Terrestrial 
Show locations on map: NO 
Show map grid: NO 
Show map scale: YES 
Show north arrow: YES 
Show plot title: YES 
Show colour scale” title: YES
Set axis labels: YES
•	Size of north arrow: 0.2
•	Size scale: 0.1
•	Scale x orientation: 0
•	Scale y orientation: 0.04
•	North arrow x orientation: 0.13
•	North arrow y orientation: 0.14
Set colour and shape of location marks: NO
Location mark size: 1
Location marks and convex hull: Add time uncertainty in years: 0
Display up to max standard error: *max*
Apply convex hull: Time-sliced spatial
Mask/Show output within range of points: 500
 
Colour palette: Pink-Yellow-Green
Reverse colors: NO 
Approximate number of colour levels: 50 
Smooth color transition: NO
Plot resolution (px): 300
Scale point size by variable: NO
Add location mark text label: NO
Axis title font size: 1
Axis label font size: 1
Center estimates: Latitude: Longitude:
Decimal places for Mean/Error: 2
Radius(km): 100

MAP
Latitude of upper left corner: 51
Longitude of upper left corner: 4
Zoom: 20

@arunge

**Comment by arunge** [comment]
@CarloCocozza Sorry for not reacting earlier! First, a question: are the additional data points coming from the input data that was used for modelling?

**Comment by CarloCocozza** [comment]
@arunge I am not 100% sure I understand what you refer to. Do you mean whether the additional data points are generated by the modelled data (although masked in my files) or the background map? If so, I believe the second is more likely. Although when I tried to remove all external nodes this had an impact on both the modelled data and geographical borders on the exported map.

**Comment by arunge** [comment]
No, actually I meant if the additional data could be part of the data that you import before modelling and the plot shows only a selected range of latitude/longitude. But sounds like it is not your data. So next we need to understand where it comes from and why it is there.

**Comment by CarloCocozza** [comment]
There could be some data that you mention just outside the map. But only for central Europe. Surely not the rest of the world.

**Comment by arunge** [comment]
@CarloCocozza With [version 24.12.1](https://github.com/Pandora-IsoMemo/DSSM/pull/261) on [DSSM beta](https://isomemoapp.com/app/dssm-beta) I integrated svg export from another R package ([Cairo](https://www.rforge.net/Cairo/)). Could you please try the export of plots as svg files again and let me know if the issue remains or if it is solved?

**Comment by CarloCocozza** [comment]
I just checked. The issue is still there it seems.

**Comment by CarloCocozza** [comment]
Not super urgent, the deadline is the end of January. Thank you and merry holidays!

**Comment by arunge** [comment]
@CarloCocozza Regarding the font:

All plots in DSSM are created with the plot function from the `graphics` package. Axis texts and the title are created with the default font that is defined with `par()`. The default font depends on the graphics device and the system (png, pdf, linux, ...)

- on Linux (our docker container uses Linux) it is often "DejaVu Sans". However, to find out the exact font, I need to add some logging into the code that prints the font while the docker container is running. I can check this together when debugging the svg issue.

Thanks for waiting and wishing you a wonderful Christmas time as well! :christmas_tree:

**Comment by arunge** [comment]
@CarloCocozza With [version 25.01.0](https://github.com/Pandora-IsoMemo/DSSM/pull/263) on [DSSM beta](https://isomemoapp.com/app/dssm-beta) the issue should be solved now. I added a clipping function to the map layers that removes map data outside of the user selected range.
Please, check if the fix is working as expected and let me know if there is something else. I am closing here.
